File: lib/test/structuredfield/parseToken.test.ts
import assert from 'node:assert';
import test from 'node:test';
import { parseToken } from '../../src/structuredfield/parse/parseToken.js';

test('parseToken', () => {
	assert.deepStrictEqual(parseToken(`*foo123/456`), { value: Symbol.for(`*foo123/456`), src: `` });
	assert.deepStrictEqual(parseToken(`foo123;456`), { value: Symbol.for(`foo123`), src: `;456` });
	assert.deepStrictEqual(parseToken(`ABC!#$%&'*+-.^_'|~:/012`), { value: Symbol.for(`ABC!#$%&'*+-.^_'|~:/012`), src: `` });
	assert.throws(() => parseToken(``), /failed to parse "" as Token/);
});

File: python/src/test/structuredfield/parseToken_test.py
import unittest

from ...structuredfield.parse.ParsedValue import ParsedValue
from ...structuredfield.parse.parseToken import parseToken
from ...structuredfield.SfException import SfException
from ...structuredfield.SfToken import SfToken


class TestParseToken(unittest.TestCase):
    def test_parseToken(self):
        token = parseToken("*foo123/456")
        self.assertEqual(token, ParsedValue(SfToken("*foo123/456"), ""))

        token = parseToken("foo123;456")
        self.assertEqual(token, ParsedValue(SfToken("foo123"), ";456"))

        token = parseToken("ABC!#$%&'*+-.^_'|~:/012")
        self.assertEqual(token, ParsedValue(SfToken("ABC!#$%&'*+-.^_'|~:/012"), ""))

        self.assertRaises(SfException, parseToken, "")
